deps/Makefile

```diff
@@ -439,8 +463,22 @@ ifeq ($(ARCH), ppc64)
 LLVM_CXXFLAGS += -mminimal-toc
 endif # ARCH == ppc64
 
+# LLVM bug #24157
+ifeq ($(USE_LLVM_SHLIB),1)
+ifeq ($(LLVM_USE_CMAKE),1)
+ifeq ($(LLVM_VER_SHORT),$(filter $(LLVM_VER_SHORT),3.3 3.4 3.5 3.6 3.7))
+$(error USE_LLVM_SHLIB=1 with LLVM_USE_CMAKE=1 requires LLVM > 3.7.x)
```

Comment:
Only tested on my local linux64 system, where it failed. Looking at the [commit](https://github.com/llvm-mirror/llvm/commit/cd9aae70e988dbd80af933bc144f1c8df431a0e9#diff-54d61d59bfb4db9f734809b8aff1dd11) where it got removed, I'd assume it would be an issue on Windows too.